xdsl/parser/attribute_parser.py

```diff
@@ -989,6 +989,83 @@ def to_type(self, parser: AttrParser, type: AnyFloat | IntegerType | IndexType):
                         parser, allow_negative=True, allow_booleans=False
                     )
 
+    def _parse_optional_int(self) -> tuple[int, Span] | None:
```

Comment:
Can you document the function to mention what it is parsing (especially the `-` part)?
Also, can you mention that this will do a lookup when parsing (i.e., it might parse `-` then rollback if the next token is not an integer).
Same for the functions below